src/tests/ie_test_utils/unit_test_utils/mocks/mock_engine/mock_plugin.cpp

```diff
@@ -6,139 +6,212 @@
 
 #include <iostream>
 #include <map>
+#include <memory>
+#include <mutex>
+#include <queue>
 #include <string>
 #include <utility>
 
+#include "cpp_interfaces/interface/ie_iplugin_internal.hpp"
 #include "description_buffer.hpp"
+#include "ie_icore.hpp"
+#include "openvino/core/except.hpp"
 #include "openvino/runtime/common.hpp"
+#include "openvino/runtime/icore.hpp"
+#include "openvino/runtime/iplugin.hpp"
+
+class MockInternalPlugin : public ov::IPlugin {
+    ov::IPlugin* m_plugin;
+    std::shared_ptr<ov::IPlugin> m_converted_plugin;
+    InferenceEngine::IInferencePlugin* m_old_plugin;
```

Comment:
should we support only new plugins?

Reply:
In caching tests we still work with legacy API. I think we can remove this support after the migration all tests to new API.
But I think this variable can be removed.